### DISCUSSION TITLE
How do you screenshot a window?

### DISCUSSION THREAD

**Discussion by joshpetit** [flameshot-org/flameshot#2060] [Q&A] [answered]
Hey! Flameshot is awesome and I've been using it for about a year now. I use i3 so don't have any native screenshot utility, so I was wondering if there is a way to capture a window in flameshot. I'm trying to have accurate window sizing since that's important for what I'm doing. If someone could show me how to do this I'd appreciate it! Thanks.

**Comment by borgmanJeremy** [comment, marked as answer]
Unfortunately it's not technically feasible to support at this time. The reason is each window manager has a different way to interact with individual windows, and qt has no abstraction for this.

Therefore we would need to implement this for each individual Linux DE as well as MacOS and Windows.


There is some proof of concept work that uses computer vision to detect window edges, but this branch is years behind the master branch.

**Reply by joshpetit** [reply]
Ok makes perfect sense! Thank you. I found a utility that let's me do it, I appreciate it!

**Reply by Midnex** [reply]
What utility may I ask?

**Reply by joshpetit** [reply]
@Midnex It's called [shutter](https://shutter-project.org/). 

I should've totally put the name. One of those "I found the answer!" -> Proceeds to not post the answer they found lol.

**Reply by Midnex** [reply]
Thanks @joshpetit

**Reply by mscurtescu** [reply]
Is this tracked as a feature request? Over time people might start contributing the platform specific bits and pieces. The feature could be enabled gradually for each platform.

Being able to take the screenshot of a window is a feature that most screenshot tools provide, because it is one of the most used ways to grab a screenshot.

**Comment by mmahmoudian** [comment]
@joshpetit since you are on X11, you can either use my code or salvage the parts that are useful for you:

https://codeberg.org/mehrad/dmenu_shot

The `Select_Window` is the part that "detects" the window and select it in Flameshot

**Reply by wd5gnr** [reply]
There is a bug about to be fixed that prevents this from working on some systems if scaling is set to something other than 1:1 but I pulled this out of @mmahmoudian's script and assigned it to Win+PrtScn in KDE Shortcuts:
```
#!/bin/bash
# get the window ID
TMP_WINDOW_ID=$(xdotool selectwindow)
unset WINDOW X Y WIDTH HEIGHT SCREEN
eval $(xdotool getwindowgeometry --shell "${TMP_WINDOW_ID}")
# Put the window in focus
xdotool windowfocus --sync "${TMP_WINDOW_ID}"
sleep 0.05
# run flameshot in gui mode in the desired coordinates
flameshot gui --region "${WIDTH}x${HEIGHT}+${X}+${Y}"
```
